### DISCUSSION TITLE
Add newer react versions to PAC please?

### DISCUSSION THREAD

**Discussion by Ionavoss** [microsoft/powerplatform-vscode#622] [General]
# PAC PCF INIT needs a revamp

Power Platform has a serious lack of UI elements. To solve this I had the idea of creating a second creator kit basically, which wraps other UI frameworks in PCF components, ready to be used. There is a problem however, and that is performance. Having many of these components on a screen can dramatically reduce performace. There is a solution to this.

## The solution
The following way of creating a react component is wonderful!
`pac pcf init -fw react`

There is only 1 problem. React for virtual components is only possible for react version 16.8.2.
Version 17 was released on Oct. 20th, 2020 and React 18 on March, 2022. Basically, our new 'virtual components', are already outdated by 3 years on react versions... too bad.

## The real problem?
The entire world moved on, converting stuff to React 17 and 18, and so have most UI frameworks done as well. It's nearly impossible to find a framework that supports react 16, and has a free license.

## My ask

Please add support for version 17 at least, this will open up plenty of PCF development projects for people, instead to being stuck on 16.

**Comment by petrochuk** [comment]
`pac pcf init` supports version of react available on [Maker Portal](https://make.powerapps.com/). As soon as Maker Portal starts supporting newer react `pac` will be updated as well.

**Reply by THEaustinlopez** [reply]
Has this happened yet?

**Comment by Ionavoss** [comment]
I understand that it's a limitation of the platform, but when will the platform get an update to support this?
It doesn't feel like there are any plans to move into the future, but its something we need to stay relevant in comparison to other platforms.

Any insights on react support in the future?
________________________________
Van: Andrew Petrochuk ***@***.***>
Verzonden: dinsdag 18 juli 2023 05:29
Aan: microsoft/powerplatform-vscode ***@***.***>
CC: Iona ***@***.***>; Author ***@***.***>
Onderwerp: Re: [microsoft/powerplatform-vscode] Add newer react versions to PAC please? (Discussion #622)


pac pcf init supports version of react available on Maker Portal<https://make.powerapps.com/>. As soon as Maker Portal starts supporting newer react pac will be updated as well.

**Comment by felixstuermer** [comment]
Any news?

**Comment by andrewmaspero** [comment]
With React 17 and 18 now the standard in the industry, isn't it time to update from the legacy React 16 dependency? 

This not only limits our ability to work with modern UI frameworks but also raises important security and performance concerns.

Additionally, I've noticed on other Microsoft repos that community engagement can feel overlooked, just like this one!

Do you see this as impacting long-term adoption and innovation on the platform, or affecting developer relations ?

A more prompt and active dialogue with developers could really help avoid potential business drawbacks by ensuring the tools meet current development needs.

From my perspective, if you have upset devs, they will just complain, and suggest customers go elsewhere as it's just too limiting, I really hope someone from the Microsoft Development team reads this. 

Could you share whether there are any concrete plans or a roadmap to modernize PAC PCF INIT?

**Comment by Suxsem** [comment]
Agree, react AND fluentui versions are way outdated...
